src/odemis/acq/feature.py

```diff
@@ -119,3 +143,312 @@ def load_project_data(path: str) -> dict:
             f.streams.value.extend(data_to_static_streams(open_acquisition(fname)))
 
     return {"overviews": overview_data, "features": features}
+
+
+def add_feature_info_filename(feature: CryoFeature, filename: str) -> str:
+    """
+    Add feature name, feature status and the counter at the end of the filename.
+    :param filename: filename given by user
+    """
+    path_base, ext = splitext(filename)
+    feature_name = feature.name.value
+    feature_status = feature.status.value
+
+    path, basename = os.path.split(path_base)
+    ptn = f"{basename}-{feature_name}-{feature_status}-{{cnt}}"
+
+    return create_filename(path, ptn, ext, count="001")
+
+# To handle the timeout error when the stage is not able to move to the desired position
+# It logs the message and raises the MoveError exception
+class MoveError(Exception):
+    pass
+
+# Time to wait for the stage to settle after moving
+STAGE_WAIT_TIME = 5 # seconds
+OBJECTIVE_WAIT_TIME = 2 # seconds
+
+class CryoFeatureAcquisitionTask(object):
```

Comment:
I think it is useful to have it be separate because you can use them individually. Since for meteor you have to move between SEM-FM, it would make sense to do all the milling then move to fm, and acquire at the features.